File: Assignments/IT Workshop/JAVA/Assignment__1__/__4__.java
class __4__ {

  static int fact(int n) {
    int f = 1;
    for (int i = 1; i < n; i++) {
      f *= i;
    }
    return f;
  }

  public static void main(String args[]) {
    int n = 6;
    System.out.println("Factorial: " + fact(n));
  }
}

File: DSA/Assignment 6/C/4.c
#include <stdio.h>
#define MAX 2
void push(int st[], int *, int);
int pop(int st[], int *);

int main()
{
    int st[MAX], top = -1, n, temp;
    printf("Enter a Number: ");
    scanf("%d", &n);
    push(st, &top, 1);
    for (int i = 2; i <= n; i++)
    {
        temp = pop(st, &top);
        push(st, &top, temp * i);
    }
    printf("Factorial: %d", pop(st, &top));
    return 0;
}

void push(int st[], int *top, int val)
{
    if (*top == MAX - 1)
        printf("Stack Overflow!");
    else
        st[++(*top)] = val;
}

int pop(int st[], int *top)
{
    if (*top == -1)
    {
        printf("Stack Underflow!");
        return -1;
    }
    else
        return st[(*top)--];
}